src/fping.c

```diff
@@ -2272,6 +2284,12 @@ int decode_icmp_ipv4(
 
     *id = icp->icmp_id;
     *seq = ntohs(icp->icmp_seq);
+    if(icp->icmp_type == ICMP_TSTAMPREPLY) {
```

Comment:
I'd think that we need to check that `reply_buf_len` is sufficiently big to contain the timestamps. Otherwise a broken implementation (or an attacker) could send a short reply packet with correct header, but without timestamps, and cause a buffer overread by `fping`.